doc/source/dev/api-dev/uarray.rst

```diff
@@ -0,0 +1,50 @@
+.. _uarray:
+
+``__ua_convert__`` in ``uarray`` backend
+=========================================
+
+SciPy supports ``uarray`` in `scipy.ndimage` and `scipy.fft`.
+Any ``uarray`` backend needs to define three protocols: ``__ua_domain__``,
+``__ua_function__`` and ``__ua_convert__``. The first two are mandatory
+and the last is optional. In this section, we provide SciPy developer
+guidelines for how the optional protocol ``__ua_convert__`` is defined
+and intended to be used, to ensure that as this protocol is added to other
+libraries making use of ``uarray``, we maintain a consistent API.
+
+
+``__ua_convert__`` is optional, but when adding ``uarray`` support in
+`scipy.ndimage` backend, we need to support and accept some arguments
+which can come in multiple ``types``. A couple of these common arguments
+in the ndimage module functions are : ``'output'`` and ``'index'``.
+
+* ``output``:
+  This is the output array which holds the output of a function.
+  It can take different types include a python ``str`` when output='f',
+  or ``output='float32'`` and dtypes like ``output=np.float32``.
```

Comment:
"This is the output array" is a bit misleading since you explain below that it can also be other types. I tried to make it a bit clearer and added one set of missing double backticks.

```suggestion
  This can be either an actual array in which the function output will be
  stored or an indication of the data type for the output array. It can take
  different types, including a python ``str`` when ``output='f'`` or
  ``output='float32'`` and dtypes like ``output=np.float32``.
```